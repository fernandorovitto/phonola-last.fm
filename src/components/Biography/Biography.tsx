import * as React from 'react';
import { ToggleButton, ToggleButtonGroup } from '@material-ui/lab';
import styled from 'styled-components';

interface PropsFromState {
  briefBio: string;
  fullBio: string;
}

export type IProps = PropsFromState;

type BioOptions = 'brief' | 'full';

interface IState {
  bioToggleOption: BioOptions;
}

class Biography extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    this.state = {
      bioToggleOption: 'brief',
    };
  }

  private _handleBio = (
    _event: React.MouseEvent<HTMLElement>,
    newBioOption: BioOptions,
  ): void => {
    if (newBioOption !== null) {
      this.setState({
        bioToggleOption: newBioOption,
      });
    }
  };

  public render(): React.ReactNode {
    return (
      <React.Fragment>
        <BioBar>
          <BioTitle>Biography</BioTitle>
          <ToggleButtonGroupStyled
            value={this.state.bioToggleOption}
            exclusive={true}
            onChange={this._handleBio}
            aria-label="text alignment"
            size={'small'}
          >
            <ToggleButton value="brief" aria-label="Brief">
              Brief
            </ToggleButton>
            <ToggleButton value="full" aria-label="Full">
              Full
            </ToggleButton>
          </ToggleButtonGroupStyled>
        </BioBar>
        <BioContainer>
          <BioContent
            dangerouslySetInnerHTML={{
              // eslint-disable-next-line @typescript-eslint/naming-convention
              __html:
                this.state.bioToggleOption === 'brief'
                  ? this.props.briefBio
                  : this.props.fullBio,
            }}
          />
        </BioContainer>
      </React.Fragment>
    );
  }
}

export default Biography;

// Styled components

const BioBar = styled('div')`
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 60px;
  justify-content: space-between;
  width: 100%;
`;
const BioTitle = styled('p')`
  font-size: 24px;
  height: 30px;
`;
const ToggleButtonGroupStyled = styled(ToggleButtonGroup)`
  background-color: white;
  height: 30px;
  margin-right: 10px;
`;

const BioContent = styled('div')`
  font-size: 16px;
  height: 100%;
  white-space: pre-line;
`;

const BioContainer = styled('div')`
  font-size: 16px;
  padding: 10px;
  white-space: pre-line;
`;
