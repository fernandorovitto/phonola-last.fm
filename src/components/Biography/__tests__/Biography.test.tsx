import 'jest-styled-components';
import Biography, { IProps } from '../Biography';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test for Biography Component', () => {
  test('Should render the component', () => {
    const props: IProps = {
      briefBio: 'Short text',
      fullBio: 'extended text',
    };
    const tree = renderer.create(<Biography {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
