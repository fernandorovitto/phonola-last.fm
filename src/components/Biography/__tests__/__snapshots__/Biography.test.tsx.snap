// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Biography Component Should render the component 1`] = `
Array [
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 60px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c1 {
  font-size: 24px;
  height: 30px;
}

.c2 {
  background-color: white;
  height: 30px;
  margin-right: 10px;
}

<div
    className="c0"
  >
    <p
      className="c1"
    >
      Biography
    </p>
    <div
      aria-label="text alignment"
      className="MuiToggleButtonGroup-root c2"
      role="group"
    >
      <button
        aria-label="Brief"
        aria-pressed={true}
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal Mui-selected MuiToggleButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="brief"
      >
        <span
          className="MuiToggleButton-label"
        >
          Brief
        </span>
      </button>
      <button
        aria-label="Full"
        aria-pressed={false}
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="full"
      >
        <span
          className="MuiToggleButton-label"
        >
          Full
        </span>
      </button>
    </div>
  </div>,
  .c1 {
  font-size: 16px;
  height: 100%;
  white-space: pre-line;
}

.c0 {
  font-size: 16px;
  padding: 10px;
  white-space: pre-line;
}

<div
    className="c0"
  >
    <div
      className="c1"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Short text",
        }
      }
    />
  </div>,
]
`;
